cirq/linalg/combinators.py

```diff
@@ -112,6 +114,31 @@ def dot(*values: Union[float, complex, np.ndarray]
     return np.linalg.multi_dot(values)
 
 
+def dot(*values: Union[float, complex, np.ndarray]
```

Comment:
While we're tweaking this method we should add a `dtype=None` parameter for forcing the output dtype.

Reply:
Do we have a use case for this?  `astype()` is an easy way to do this, I'd say wait until you see that this is commonly needed?

Reply:
k